From 1656649fd3ae18e3cf1e08418a6a2fb7e2389607 Mon Sep 17 00:00:00 2001
From: nikhilad <nikhilad@ami.com>
Date: Tue, 25 Nov 2025 20:30:05 +0530
Subject: [PATCH] Reverting the patch for Preserve config failure

Throwing an error While updating the
preserve config properties as Invalid
argument
So, Reverting the community change
which is causing the issue

Signed-off-by: nikhilad <nikhilad@ami.com>
---
 src/entity_manager/entity_manager.cpp | 26 ++++++++++++++------------
 1 file changed, 14 insertions(+), 12 deletions(-)

diff --git a/src/entity_manager/entity_manager.cpp b/src/entity_manager/entity_manager.cpp
index 8204082..904b731 100644
--- a/src/entity_manager/entity_manager.cpp
+++ b/src/entity_manager/entity_manager.cpp
@@ -51,7 +51,8 @@ constexpr const char* tempConfigDir = "/tmp/configuration/";
 constexpr const char* lastConfiguration = "/tmp/configuration/last.json";
 
 static constexpr std::array<const char*, 7> settableInterfaces = {
-    "FanProfile", "Pid", "Pid.Zone", "Stepwise", "Thresholds", "Polling", "Preserve"};
+    "FanProfile", "Pid",     "Pid.Zone", "Stepwise",
+    "Thresholds", "Polling", "Preserve"};
 
 // NOLINTBEGIN(cppcoreguidelines-avoid-non-const-global-variables)
 
@@ -234,6 +235,16 @@ void postToDbus(const nlohmann::json& newConfiguration,
                     objServer, getPermission(itemType));
             }
 
+            std::shared_ptr<sdbusplus::asio::dbus_interface> itemIface =
+                dbus_interface::createInterface(
+                    objServer, ifacePath,
+                    "xyz.openbmc_project.Configuration." + itemType,
+                    boardNameOrig);
+
+            dbus_interface::populateInterfaceFromJson(
+                systemConfiguration, jsonPointerPath, itemIface, item,
+                objServer, getPermission(itemType));
+
             for (const auto& [name, config] : item.items())
             {
                 jsonPointerPath = jsonPointerPathBoard;
@@ -304,16 +315,6 @@ void postToDbus(const nlohmann::json& newConfiguration,
                 }
             }
 
-            std::shared_ptr<sdbusplus::asio::dbus_interface> itemIface =
-                dbus_interface::createInterface(
-                    objServer, ifacePath,
-                    "xyz.openbmc_project.Configuration." + itemType,
-                    boardNameOrig);
-
-            dbus_interface::populateInterfaceFromJson(
-                systemConfiguration, jsonPointerPath, itemIface, item,
-                objServer, getPermission(itemType));
-
             topology.addBoard(boardPath, boardType, boardNameOrig, item);
         }
 
@@ -671,7 +672,8 @@ int main()
     });
     dbus_interface::tryIfaceInitialize(entityIface);
 
-    if (em_utils::fwVersionIsSame() || std::filesystem::exists(configuration::currentConfiguration))
+    if (em_utils::fwVersionIsSame() ||
+        std::filesystem::exists(configuration::currentConfiguration))
     {
         if (std::filesystem::is_regular_file(
                 configuration::currentConfiguration))
-- 
2.34.1

